metrics/collector.go

```diff
@@ -41,17 +49,20 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	// There's seemingly an iptables bug where occasionally all counters drop to 0, but only for a split second,
-	// and then they jump back up. This really upsets prometheus, so we should drop scrapes when everything is zero.
-	var zeroCounter int
+	// There's seemingly an iptables bug where occasionally all counters drop to 0 or near 0, but only for a split second,
+	// and then they jump back up. This really upsets prometheus, so we should drop scrapes when everything is small.
+	var scrapeAllowed bool
 	for _, result := range results {
-		if result.PacketCount == 0 {
-			zeroCounter++
+		if result.PacketCount >= c.minCounter {
```

Comment:
Are we sure we never want to emit metrics when they're e.g. all zeroes?

Reply:
The only way they can all be zeroes is if essentially no traffic is going to the node. There is one unfortunate case where there is only one workload on the node, which receives no traffic, and then it suddenly starts getting some traffic (dropped or otherwise); we wouldn't get metrics for this until it receives its nth SYN packet. I don't see this as a big problem